Source/Core/VideoBackends/Software/Tev.cpp

```diff
@@ -812,28 +812,28 @@ void Tev::SetRegColor(int reg, int comp, bool konst, s16 color)
 
 void Tev::DoState(PointerWrap &p)
 {
-	p.DoArray(Reg, sizeof(Reg));
+	p.DoArray(Reg);
 
-	p.DoArray(KonstantColors, sizeof(KonstantColors));
-	p.DoArray(TexColor,4);
-	p.DoArray(RasColor,4);
-	p.DoArray(StageKonst,4);
-	p.DoArray(Zero16,4);
+	p.DoArray(KonstantColors);
```

Comment:
Yes it should. The original code was a bug anyways, since s16[4][4] has an overall byte size of 32, not 256 like the original code would produce (since the type of T* would be short[4] at that point, which would make the size calculation 32 * 8)